cdc/kv/region_state.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kv
+
+import (
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pingcap/tiflow/pkg/regionspan"
+	"github.com/tikv/client-go/v2/tikv"
+)
+
+const (
+	minRegionStateBucket = 4
+	maxRegionStateBucket = 16
+)
+
+type singleRegionInfo struct {
+	verID        tikv.RegionVerID
+	span         regionspan.ComparableSpan
+	checkpointTs uint64
+	rpcCtx       *tikv.RPCContext
+}
+
+func newSingleRegionInfo(
+	verID tikv.RegionVerID,
+	span regionspan.ComparableSpan,
+	ts uint64,
+	rpcCtx *tikv.RPCContext,
+) singleRegionInfo {
+	return singleRegionInfo{
+		verID:        verID,
+		span:         span,
+		checkpointTs: ts,
+		rpcCtx:       rpcCtx,
+	}
+}
+
+type regionFeedState struct {
+	sri       singleRegionInfo
+	requestID uint64
+	stopped   int32
+
+	lock           sync.RWMutex
```

Comment:
Maybe we can use atomic values to avoid using the lock if the lock contention is really that severe.

Reply:
Atomic value access also has some cost, maybe it's less than locks.

Since there are so many fields need to be accessed exclusively, so maybe a `RWMutex` shared by all fields is ok.